pkg/network/http/ebpf_ssl.go

```diff
@@ -198,37 +358,38 @@ func (o *sslProgram) Stop() {
 	o.perfHandler.Stop()
 }
 
-func addHooks(m *manager.Manager, probes map[string]string) func(string) error {
+func addHooks(m *manager.Manager, probes []manager.ProbesSelector) func(string) error {
 	return func(libPath string) error {
 		uid := getUID(libPath)
-		for sec, funcName := range probes {
-			p, found := m.GetProbe(manager.ProbeIdentificationPair{
-				EBPFSection:  sec,
-				EBPFFuncName: funcName,
-				UID:          uid,
-			})
-			if found {
-				if !p.IsRunning() {
-					err := p.Attach()
-					if err != nil {
-						return err
+
+		for i := range probes {
+			for _, selector := range probes[i].GetProbesIdentificationPairList() {
+				identifier := manager.ProbeIdentificationPair{
+					EBPFSection:  selector.EBPFSection,
+					EBPFFuncName: selector.EBPFFuncName,
+					UID:          uid,
+				}
+				probes[i].EditProbeIdentificationPair(selector, identifier)
+				p, found := m.GetProbe(identifier)
+				if found {
+					if !p.IsRunning() {
+						err := p.Attach()
```

Comment:
sowatcher would probably need some change if SSL_read/write_ex() doesn't exist, can't be binded (ex: openssl 1.0)
But this could be addressed in different PR

Reply:
I'm using the Selectors and marked `SSL_{read, write}_ex` methods as "BestEffort" for that purpose

Reply:
resolving as we spoke about that in person